OpenIndiana/oi-userland
# library/perl-5/net-ssleay: perl 5.34 rebuild

rebuild library/perl-5/net-ssleay for perl 5.34 and modernize `Makefile` and the manifest.

**Question:** the manifest records a dependency on zlib: `depend fmri=library/zlib@1.2.3-0.169 type=require`.  I did not alter this, and I'm not sure if I should or not.  I believe that our current `library/zlib@1.2.11` satisfies the stated dependency (I think; I could be wrong), so I think it's safe to leave it as it is, but I wanted to attention to it, in case it should be changed.

This doesn't need any other particular perl modules to be packaged first, and you've already merged PR #7512 , so that requirement is met too.

Other changes were straightforward:

`Makefile`:
1.  add `BUILD_STYLE=makemaker` and `BUILT_BITS=32_and_64`.  Add a comment about `USE_OPENSSL11`, but that's not an option until this module is updated to a later version.
2. bump `COMPONENT_REVISION`
3. add `COMPONENT_FMRI`, `COMPONENT_SUMMARY`, and `COMPONENT_CLASSIFICATION` using the values from the manifest.
4. update `COMPONENT_PROJECT_URL` and `COMPONENT_ARCHIVE_URL` for https and current locations
5. drop `COMPONENT_BUGDB`.  OI doesn't use this.
6. add `COMPONENT_LICENSE=Artistic`.  The old value from the `net-ssleay-PERLVER.p5m` was outdated.
7. set `PERL_VERSIONS` and `PERL_64_ONLY_VERSIONS`, overriding the current default to include 5.34.
8. drop the existing macro includes and switch to including `common.mk`
9. drop `COMPONENT_TEST_TARGET=test`.  Not needed since `makemaker.mk` was updated.
10. drop `build/install/test`.  The targets specified in `makemaker.mk` work.
11.  add `COMPONENT_TEST_MASTER` and specify a per-version (but not per-version & per-bits) master file for the tests.  The results do differ between versions, so we can't use one `results-all.master`
12. add suitable `COMPONENT_TEST_TRANSFORMS`
13. update `REQUIRED_PACKAGES`
14. run `gmake sample-manifest` and include the updated `manifests/sample-manifest.p5m`

`net-ssleay-PERLVER.p5m`:
1. reference `$(COMPONENT_FMRI)`, now specified in the `Makefile`
2. reference `$(COMPONENT_SUMMARY)` from `Makefile`
3. reference `$(COMPONENT_CLASSIFCIATION)` from `Makefile`
4. drop `set name=org.opensolaris.arc-caseid value=PSARC/2009/189`, OI doesn't use this
5. update the `license` line to reference the default `$(COMPONENT_LICENSE_FILE)` and `$(COMPONENT_LICENSE)`
6. add a runtime dependency so that each versioned modules picks up a dependency on its version of perl
7. add a runtime dependency so that each versioned module also depends upon the unversioned `library/perl-5/net-ssleay`

`patches`:
1. add a new patch, suitable for all 3 versions but only required for the perl 5.34 module build.  perl 5.26 and later no longer include `.` in the `@INC` module search path, so the `Makefile.PL` was not able to find some internal modules specified via a relative path.

`Net-SSLeay.license`:
1. add this based on the license file shipped with this version, and drop the previous `net-ssleay.copyright` file.  As mentioned above, the license changed previously and our package didn't catch it.
